clap_generate/src/generators/shells/fish.rs

```diff
@@ -107,8 +114,24 @@ fn gen_fish_inner(root_command: &str, app: &App, buffer: &mut String) {
         buffer.push('\n');
     }
 
+    let mut subcommand_template = basic_template;
+    // We only need to do this for subcommands of subcommands
+    if !parent_commands.is_empty() {
+        // Removes the `"` at the end of the basic_template
+        subcommand_template.remove(subcommand_template.len() - 1);
```

Comment:
I would prefer if we don't do it like this and instead add `"`

Reply:
Actually I solved both your issues and a third one by applying the "not sub_subcommand" to the basic template at the beginning, because this should does not only apply to subcommands but also options.